### PR TITLE
DAMD-89: Document and implement DetectorMap

### PR DESCRIPTION
**Pull request by PaulPrice**

### CODE REVIEW

File: python/pfs/datamodel/pfsDetectorMap.py [hunk 1]
```diff
@@ -0,0 +1,708 @@
...
+        x : array-like
+            Positions at which to evaluate the spline.
+
+        Returns
+        -------
+        y : array-like
+            Spline values.
+        """
+        return self._impl(x)
+
```
Comment (hassanxp) [inline python/pfs/datamodel/pfsDetectorMap.py:139]: Could these (in the future) be moved to `pfs_utils`?

File: python/pfs/datamodel/pfsDetectorMap.py [hunk 2]
```diff
@@ -0,0 +1,708 @@
...
+        identity : `pfs.datamodel.CalibIdentity`
+            Identity of the data of interest.
+        """
+        dirName, fileName = os.path.split(path)
+        matches = re.search(cls.filenameRegex, fileName)
+        if not matches:
+            raise RuntimeError("Unable to parse filename: %s" % (fileName,))
+        identity = matches.groupdict()
+        identity["obsDate"] = None  # Not exposed in the detectorMap filename
+        return CalibIdentity.fromDict()
```
Comment (hassanxp) [inline python/pfs/datamodel/pfsDetectorMap.py:215]: The `identity` doesn't look as if it is passed to `CalibIdentity`. Is this fixed in later commits?

File: datamodel.txt [hunk 3]
```diff
@@ -190,6 +190,67 @@ The PDU has keys to define which data were used to construct the FiberTrace:
...
+        fiberId             32-bit int      Fiber identifier (primary key)
+        spatialOffsets      32-bit float    Offset in spatial dimension (x)
+        spectralOffsets     32-bit float    Offset in spectral dimension (y)
+
+The COEFFICIENTS table contains:
+        x                   32-bit float    x coefficients
+        y                   32-bit float    y coefficients
+
+The RIGHTCCD table contains:
+        coeff               32-bit float    Affine transform coefficients
```
Comment (hassanxp) [inline datamodel.txt:251]: Shouldn't this be `coeffs` (plural) for consistency? The description should mention 6 coefficients for a 2D transformation, just to make it more concrete.

**Review by hassanxp — APPROVED** [review]

File: python/pfs/datamodel/pfsDetectorMap.py (hunk 1 above)
Comment (PaulPrice) [reply python/pfs/datamodel/pfsDetectorMap.py:139]: Yes, but I don't know what that would gain.

File: python/pfs/datamodel/pfsDetectorMap.py (hunk 2 above)
Comment (PaulPrice) [reply python/pfs/datamodel/pfsDetectorMap.py:215]: Fixed.

File: datamodel.txt (hunk 3 above)
Comment (PaulPrice) [reply datamodel.txt:251]: I'm inclined to leave the name, as it affects data on disk.

**Pull request merged**